scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -609,6 +611,37 @@ class ScioContext private[scio] (val options: PipelineOptions,
   }
 
   /**
+    * Get an SCollection for a Pub/Sub subscription that includes message attributes.
+    * @group input
+    */
+  def pubsubSubscriptionWithAttributes[T: ClassTag](sub: String,
+                                      idLabel: String = null,
+                                      timestampLabel: String = null)
+  : SCollection[(T, Map[String, String])] = requireNotClosed {
+    if (this.isTest) {
+      this.getTestInput(PubsubIO(sub))
+    } else {
+      val elementCoder = pipeline.getCoderRegistry.getScalaCoder[T]
+      val outputCoder  = pipeline.getCoderRegistry.getScalaCoder[(T, Map[String, String])]
+      val parseFn = Functions.simpleFn { msg: PubsubMessage =>
+        val element = CoderUtils.decodeFromByteArray(elementCoder, msg.getMessage)
+        val attributes = msg.getAttributeMap.asScala.toMap
```

Comment:
The output type of `parseFn` is what's used as the output type of the read transform:
```
public Read<T> withAttributes(SimpleFunction<PubsubMessage, T> parseFn)
```
and thus the element type of the returned SCollection. Would casting the wrapper to an instance of `Map[String, String]` be the way to go?